docs/developer/update-v1alpha1-provider-to-v1alpha2-infra-provider.md

```diff
@@ -0,0 +1,188 @@
+| [`sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.Actuator`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e6a57dc61826b8c7806eba22a513c9722c420754/pkg/cloud/aws/actuators/machine/actuator.go#L57-L65) | [`sigs.k8s.io/cluster-api-provider-aws/pkg/controller/awsmachine.ReconcileAWSMachine`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awsmachine/awsmachine_controller.go#L104-L109) |
+
+### The provider cluster controller
+
+Instead of processing the CAPI `Cluster` resources like the old actuator model, the new provider cluster controller operates on the new provider `Cluster` CRD. However, the overall workflow should feel the same as the old cluster actuator. For example, take the AWS cluster controller's [reconcile function](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L105-L162), it:
+
+1. Fetches the [AWS cluster resource](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L113-L121):
+
+    ```golang
+    awsCluster := &infrastructurev1alpha2.AWSCluster{}
+    err := r.Get(ctx, request.NamespacedName, awsCluster)
+    if err != nil {
+        if apierrors.IsNotFound(err) {
+            return reconcile.Result{}, nil
+        }
+        return reconcile.Result{}, err
+    }
+    ```
+
+2. [Fetches the CAPI cluster resource](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L125-L133) that has a one-to-one relationship with the AWS cluster resource:
+
+    ```golang
+    cluster, err := util.GetOwnerCluster(ctx, r.Client, awsCluster.ObjectMeta)
+    if err != nil {
+        return reconcile.Result{}, err
+    }
+    if cluster == nil {
+        logger.Info("Waiting for Cluster Controller to set OwnerRef on AWSCluster")
+        return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+    }
+    ```
+
+    If the AWS cluster resource does not have a corresponding CAPI cluster resource then the reconcile operation is requeued until such time the relationship is established.
+
+3. Uses a `defer` statement to [ensure the AWS and CAPI cluster resources are always patched](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L148-L153) back to the API server:
```

Comment:
```suggestion
3. Uses a `defer` statement to [ensure the AWSCluster resource is always patched](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L148-L153) back to the API server:
```

Only the infrastructure provider defined resource should be updated by the infrastructure provider controllers, the related objects should only be used for reference